GitWorkflow/GitWorkflow/Player.cpp

```diff
@@ -0,0 +1,77 @@
+#include "Player.hpp"
+
+Player::Player(const sf::Vector2f& position, const sf::Vector2f& size, const sf::Color& color)
+	: Drawable(), m_model(size)
+{
+	m_model.setFillColor(color);
+	m_model.setPosition(position);
+}
+
+void Player::move(const sf::Vector2f& offset)
+{
+	m_model.move(offset);
+}
+
+void Player::setPosition(const sf::Vector2f& position)
+{
+	m_model.setPosition(position);
+}
+
+const sf::Vector2f& Player::getPosition() const
+{
+	return m_model.getPosition();
+}
+
+void Player::setVelocity(const sf::Vector2f& velocity)
+{
+	m_velocity = velocity;
+}
+
+const sf::Vector2f& Player::getVelocity() const
+{
+	return m_velocity;
+}
+
+const sf::FloatRect Player::getGlobalBounds() const
+{
+	return m_model.getGlobalBounds();
+}
+
+void Player::collide(const Platform& platform)
+{
+	auto intersection = util::intersection(getGlobalBounds(), platform.getGlobalBounds());
+	bool verticalCollision = abs(intersection.width) > abs(intersection.height);
+
+	move(verticalCollision ? sf::Vector2f(0.0f, intersection.height) : sf::Vector2f(intersection.width, 0.0f));
+
+	if (verticalCollision)
+	{
+		if(m_velocity.y > 0.0f) m_hasJumped = false;
+		setVelocity({ m_velocity.x, 0.0f });
+	}
```

Comment:
If falling alongside an edge and pushing towards it, the last frame of collision could stop the player's y-velocity.
This needs some refinement to check that the player is above the platform.
If the collision is not vertical, the x-velocity should be reset.

Reply:
Someone else can fix it.